### PR TITLE
Updates to allow for nested objects as data/keys

### PR DESCRIPTION
**Pull request by minimalmonkey**
This PR is related to issue #57 

- Updated `format-message-interpret` to allow for nested data objects
- Updated `eslint-plugin-format-message` to check for nested data objects

This update will allow `MessageFormat` to be used with nested data objects, for example:
```js
var pattern = 'x{ arg.y }z'
var message = new MessageFormat('en', pattern).format({ arg: { y: 'y' } })
//-> 'xyz'
```

In order to make this change backwards compatible, arguments with periods in the name are still supported and will be given preference over nested data, for example:
```js
var pattern = 'x{ arg.y }z'
// the below will favor the first key "arg.y" over the nested object
var message = new MessageFormat('en', pattern).format({ 'arg.y': 'y', arg: { y: 'a' } })
//-> 'xyz'
```

- Still to do for complete support is updating the babel plugin to also support this

### CODE REVIEW

File: packages/format-message-interpret/index.js [hunk 1]
```diff
@@ -138,6 +138,21 @@ function interpretSelect (locale, id, children) {
 
 function interpretSimple (id) {
   return function format (args) {
-    return '' + args[id]
+    return '' + getArg(id, args)
+  }
+}
+
+function getArg (id, args) {
+  if (args[id] !== undefined) return args[id]
```
Comment (thetalecrafter) [inline packages/format-message-interpret/index.js]: Would the `in` operator be better here? `if (id in args)`, or how should `null`s be handled?

What should the following do?
```
__('{ x.y }', { 'x.y': null, x: { y: 'y' } })
```

**Comment by thetalecrafter** [comment]
This is the first time I've actually used the review feature on github. :)

I don't think extract would need any changes, other than tests to validate it does work correctly.
Transform hopefully only needs changes in the `inline-message.js`.

File: packages/format-message-interpret/index.js (hunk 1 above)
Comment (minimalmonkey) [inline packages/format-message-interpret/index.js]: Yeah we can use `if (id in args)` and yes, that's a good question. I would say in the example if `'x.y'` is set to anything it'll be used, so `null` will be returned. I'd avoid checking for falsy values to keep things simple and I think we could maybe assume that users of the library will be either using periods in var names or nested objects but unlikely a mixture of both. What are your thoughts? I guess the alternative would be to check for `undefined` or `null`. Happy to go for whichever you prefer here. I'd be leaning towards using `if (id in args)` though and can update that if you agree.

Comment (thetalecrafter) [inline packages/format-message-interpret/index.js]: I think that's a good way to go. There are probably arguments to ignore `null` and `undefined` values, but I think the `in` check is simple and reasonable. Let's go with that.

**Comment by zdfs** [comment]
Hey @thetalecrafter (Andy), the one thing we haven't done yet is add the Babel support. We kind of got stuck on it and was wondering if you could write it faster? I _think_ it's the only thing left to do. I'm happy to give it another go, but to be honest, I'm not super familiar with Babel and could use all the help I can get. 🙇

**Comment by thetalecrafter** [comment]
No worries. This is an excellent addition, and I can probably carve out a little time to finish the babel portion.

**Comment by minimalmonkey** [comment]
This is great! Thanks so much @thetalecrafter - really appreciate all the work you put into this library.

**Pull request merged**